src/gui/speedlimitdialog.cpp

```diff
@@ -50,58 +65,111 @@ SpeedLimitDialog::~SpeedLimitDialog()
+void SpeedLimitDialog::updateAltUploadSpinValue(const int value)
+{
+    m_ui->spinAltUploadLimit->setValue(value);
+}
 
-        const int val = dlg.getSpeedLimit();
-        if (val < 0)
-            return 0;
-        return (val * 1024);
-    }
+void SpeedLimitDialog::updateDownloadSliderValue(const int value)
+{
+    if (value > m_ui->sliderDownloadLimit->maximum())
+        m_ui->sliderDownloadLimit->setMaximum(value);
+    m_ui->sliderDownloadLimit->setValue(value);
+}
 
-    return -2;
+void SpeedLimitDialog::updateUploadSliderValue(const int value)
+{
+    if (value > m_ui->sliderUploadLimit->maximum())
+        m_ui->sliderUploadLimit->setMaximum(value);
+    m_ui->sliderUploadLimit->setValue(value);
 }
 
-void SpeedLimitDialog::updateSpinValue(const int value)
+void SpeedLimitDialog::updateAltDownloadSliderValue(const int value)
 {
-    m_ui->spinBandwidth->setValue(value);
+    if (value > m_ui->sliderAltDownloadLimit->maximum())
+        m_ui->sliderAltDownloadLimit->setMaximum(value);
+    m_ui->sliderAltDownloadLimit->setValue(value);
 }
 
-void SpeedLimitDialog::updateSliderValue(const int value)
+void SpeedLimitDialog::updateAltUploadSliderValue(const int value)
 {
-    if (value > m_ui->bandwidthSlider->maximum())
-        m_ui->bandwidthSlider->setMaximum(value);
-    m_ui->bandwidthSlider->setValue(value);
+    if (value > m_ui->sliderAltUploadLimit->maximum())
+        m_ui->sliderAltUploadLimit->setMaximum(value);
+    m_ui->sliderAltUploadLimit->setValue(value);
 }
 
-int SpeedLimitDialog::getSpeedLimit() const
+int SpeedLimitDialog::getDownloadSpeedLimit() const
 {
-    return m_ui->spinBandwidth->value();
+    return m_ui->spinDownloadLimit->value();
 }
 
-void SpeedLimitDialog::setupDialog(long maxSlider, long val)
+int SpeedLimitDialog::getUploadSpeedLimit() const
 {
-    val = qMax<long>(0, val);
+    return m_ui->spinUploadLimit->value();
+}
 
-    if (maxSlider <= 0)
-        maxSlider = 10000;
+int SpeedLimitDialog::getAltDownloadSpeedLimit() const
+{
+    return m_ui->spinAltDownloadLimit->value();
+}
 
-    // This can happen for example if global rate limit is lower
-    // than torrent rate limit.
-    if (val > maxSlider)
-        maxSlider = val;
+int SpeedLimitDialog::getAltUploadSpeedLimit() const
+{
+    return m_ui->spinAltUploadLimit->value();
+}
 
-    m_ui->bandwidthSlider->setMaximum(maxSlider);
-    m_ui->bandwidthSlider->setValue(val);
-    updateSpinValue(val);
+void SpeedLimitDialog::initDialog()
+{
+    const int uploadVal = qMax(0, (BitTorrent::Session::instance()->globalUploadSpeedLimit() / 1024));
+    const int downloadVal = qMax(0, (BitTorrent::Session::instance()->globalDownloadSpeedLimit() / 1024));
+    const int maxUpload = qMax(10000, (BitTorrent::Session::instance()->globalUploadSpeedLimit() / 1024));
+    const int maxDownload = qMax(10000, (BitTorrent::Session::instance()->globalDownloadSpeedLimit() / 1024));
+    m_ui->sliderUploadLimit->setMaximum(maxUpload);
```

Comment:
I would suggest we use a helper function here:
```c++
{
    const auto updateSlider = [](QSlider &slider, const int value, const int maximum)
    {
        slider->setValue(value);
        slider->setMaximum(maximum);
    };

    const int uploadVal = qMax(0, (BitTorrent::Session::instance()->globalUploadSpeedLimit() / 1024));
    // const int downloadVal ...
    updateSlider(m_ui->sliderUploadLimit, uploadVal, maxUpload);
    // updateSlider(...
}
```

Reply:
Done thanks.